metaseq/models/transformer.py

```diff
@@ -746,16 +741,17 @@ def buffered_future_mask(self, tensor, input_tokens=None):
                     ] = float("-inf")
 
             if self.use_alibi:
-                alibi = self.alibi.repeat(batch_size, 1, 1)  # batch_size, 1, 1
-                self._future_mask = self._future_mask.unsqueeze(0) + alibi
+                self._future_mask = (
+                    self._future_mask.unsqueeze(0)
+                    .expand(self.args.decoder_attention_heads, max_seq_len, max_seq_len)
+                    .contiguous()
```

Comment:
i wonder why the contiguous is necessary

let's make sure to benchmark speed too!